collector/eventhub/src/main/java/zipkin/collector/eventhub/IEventProcessorHostFactory.java

```diff
@@ -0,0 +1,24 @@
+/**
+ * Copyright 2017 The OpenZipkin Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package zipkin.collector.eventhub;
+
+public interface IEventProcessorHostFactory {
```

Comment:
For example, if you look at the Kafka, which is very similar in purpose, you'll notice we don't expose wide-open hooks like this intentionally. In almost (if not) all code in zipkin, we avoid boilerplate such as creating interfaces and then identical implementations (because we don't expose the types public in the first place).

One alternative to passing 7 paramters (and next version 8 or 9) is to use a builder, similar to how we do in Kafka
https://github.com/openzipkin/zipkin/blob/master/zipkin-collector/kafka/src/main/java/zipkin/collector/kafka/KafkaCollector.java#L188

as long as the fields are saved immutable (final and copied if the source is mutable), all good